### ISSUE TITLE
Fok 1.42.6 - Incorrect "changed files" detection?

### ISSUE THREAD

**Issue by jgodfrey** [ForkIssues/TrackerWin#604]
For a while now, I've been noticing some strange "change detection" - on multiple PC systems. 

It's always a case where a file is shown in the "Unstaged Changes" section, but when the file is selected, nothing is shown in the diff panel. Oddly, even when I activate "show entire file" in the diff panel, nothing is shown in the panel (for a file that **definitely** has contents).

Additionally, if I stage such a file, it (probably correctly?) never appears in the staged panel - it just disappears - as if it didn't actually contain any changes in the first place.

Below is an image of a file in such a state (note that "show entire file" is active, but nothing is displayed). As noted above, when I stage this file, it's simply removed from the "changed" list...

![image](https://user-images.githubusercontent.com/1459347/71679378-136f1480-2d4d-11ea-9182-d27544c1478c.png)

**Comment by DanPristupov** [comment]
I suppose your text editor and your repository use different EOL symbols.

https://stackoverflow.com/questions/2825428/why-should-i-use-core-autocrlf-true-in-git

UPD: May be you should open terminal in Fork (Ctrl+Alt+T) and enter `git config --global core.autocrlf true`

**Comment by jgodfrey** [comment]
Hmmm... I'll look into that, but I'm not sure that's the issue here. Could that also explain the "diff panel shows nothing, even when set to show entire file contents" issue? And, the "file shown in unstaged panel, but disappears after staging" issue?

**Comment by DanPristupov** [comment]
> Could that also explain the "diff panel shows nothing, even when set to show entire file contents" issue?

This might be caused by a chmod or encoding (Unicode -> ANSI) change. Open terminal and check `git status`.

> And, the "file shown in unstaged panel, but disappears after staging" issue?

Auto-CRLF is a feature which changes CR->LF automatically *on stage*. So, yes, it can. However it's also possible that the file has a mode which isn't supported on Windows.

**Comment by jgodfrey** [comment]
I just had this happen again, so I did some additional investigation. As you guessed, it does seem to be caused by EOL differences (Win / Unix). I'll verify the other PC / code base where I've seen it too (when it happens again), but I assume it's the same issue.

Thanks for your input. 

That said, I still don't understand why I can't *see* the file contents in Fork's diff view when "show entire file" is enabled. That seems wrong to me.

**Comment by DanPristupov** [comment]
> That said, I still don't understand why I can't see the file contents in Fork's diff view when "show entire file" is enabled.

This is an internal limitation. The reason is that `git status -- /path/to/filename` doesn't return the content of the file. However, it's a good point, Fork might return it that case something which makes at least some sense :).

**Comment by jgodfrey** [comment]
> However, it's a good point, Fork might return it that case something which makes at least some sense :).

Yep, agreed. Placing a file in the "Unstaged Changed" list, but then not showing me *any* changes (or even the file contents in the described case) is confusing from a User-perspective. 

Based on this thread, I understand the issue now, but it'd be great if the situation could be better clarified within Fork.
